charts/opentelemetry-collector/templates/_config.tpl

```diff
@@ -56,7 +60,7 @@ Build config file for deployment OpenTelemetry Collector
 {{- end }}
 
 {{- define "opentelemetry-collector.applyHostMetricsConfig" -}}
-{{- $config := mustMergeOverwrite (include "opentelemetry-collector.hostMetricsConfig" .Values | fromYaml) .config }}
+{{- $config := mustMergeOverwrite (dict "service" (dict "pipelines" (dict "metrics" (dict "receivers" list)))) (include "opentelemetry-collector.hostMetricsConfig" .Values | fromYaml) .config }}
```

Comment:
why did these need to change?

Reply:
Helm can't handle appending to empty lists.  This additional merge ensures the required list is present. 

This previously wasn't an issue bc the default config defined lists for each pipeline component type, but with the new alternative config, it would be possible to define a logs pipeline, only set the processors and exporters lists, and then enable the `logsCollection` preset.  In order for the preset to insert the filelogreceiver, it also needs to addd the list.

Reply:
Why don't we just leave the alternateConfig (or configOverride) as is and say that's the end result and no presets are applied? It'd make things easy for us and users I think.  Do you think it's pretty common that users would want their own config + presets?

Reply:
I included the feature bc it is a common use case to enable presets when using the chart as a subchart.